src/pyscaffold/update.py

```diff
@@ -0,0 +1,272 @@
+            rule == FileOp.NO_CREATE or
+            path_exists(path) and rule == FileOp.NO_OVERWRITE)
+
+    if skip:
+        logger.report('skip', path)
+        return None
+
+    return content
+
+
+def read_setupcfg(project_path):
+    """Reads-in setup.cfg for updating
+
+    Args:
+        project_path (str): path to project
+
+    Returns:
+
+    """
+    path = join_path(project_path, 'setup.cfg')
+    updater = ConfigUpdater()
+    updater.read(path)
+    return updater
+
+
+def invoke_action(action, struct, opts):
+    """Invoke action with proper logging.
+
+    Args:
+        struct (dict): project representation as (possibly) nested
+            :obj:`dict`.
+        opts (dict): given options, see :obj:`create_project` for
+            an extensive list.
+
+    Returns:
+        tuple(dict, dict): updated project representation and options
+    """
+    logger.report('invoke', helpers.get_id(action))
+    with logger.indent():
+        struct, opts = action(struct, opts)
+
+    return struct, opts
+
+
+def get_curr_version(project_path):
+    """Retrieves the PyScaffold version that put up the scaffold
+
+    Args:
+        project_path: path to project
+
+    Returns:
+        Version: version specifier
+    """
+    setupcfg = ConfigParser()
+    setupcfg.read(join_path(project_path, 'setup.cfg'))
+    return parse_version(setupcfg['pyscaffold']['version'])
+
+
+def version_migration(struct, opts):
+    """Migrations from one version to another
+
+    Args:
+        struct (dict): previous directory structure (ignored)
+        opts (dict): options of the project
+
+    Returns:
+        tuple(dict, dict):
+            structure as dictionary of dictionaries and input options
+    """
+    update = opts.get('update')
+
+    if not update:
+        return struct, opts
+
+    curr_version = get_curr_version(opts['project'])
+
+    # specify how to migrate from one version to another as ordered list
+    migration_plans = [
+        (parse_version('3.1'), [add_entrypoints,
+                                add_setup_requires])
+    ]
+    for plan_version, plan_actions in migration_plans:
+        if curr_version < plan_version:
+            struct, opts = reduce(lambda acc, f: invoke_action(f, *acc),
+                                  plan_actions, (struct, opts))
+
+    # note the updating version in setup.cfg for future use
+    update_pyscaffold_version(opts['project'], opts['pretend'])
+    # replace the old version with the updated one
+    opts['version'] = pyscaffold_version
+    return struct, opts
+
+
+def add_entrypoints(struct, opts):
+    """Add [options.entry_points] to setup.cfg
+
+    Args:
+        struct (dict): previous directory structure (ignored)
+        opts (dict): options of the project
+
+    Returns:
+        tuple(dict, dict):
+            structure as dictionary of dictionaries and input options
+    """
+    setupcfg = read_setupcfg(opts['project'])
+    section_str = """[options.entry_points]
```

Comment:
Maybe we should add this to a template file and also make the setup.cfg template to use this one?

Reply:
Yes, good point, will do that!